Tests/kaas/k8s-version-policy/k8s_version_policy.py

```diff
@@ -479,63 +487,40 @@ async def main(argv):
     async with aiohttp.ClientSession(connector=connector) as session:
         cve_affected_ranges = await collect_cve_versions(session)
 
-    contexts = ["stable", "oldstable", "oldoldstable", "oldoldoldstable"]
-    seen_branches = set()
-
     try:
-        for context in contexts:
-            logger.info("Checking cluster of kubeconfig context '%s'.", context)
-            cluster = await get_k8s_cluster_info(config.kubeconfig, context)
-            cluster_branch = cluster.version.branch
-            seen_branches.add(cluster_branch)
-
-            # allow older k8s branches, but not for the first context (stable)
-            allow_older = context != contexts[0]
-
-            if check_k8s_version_recency(cluster.version, cve_affected_ranges, allow_older):
-                logger.info(
-                    "The K8s cluster version %s of cluster '%s' is still in the recency time window.",
-                    cluster.version,
-                    cluster.name,
-                )
-            else:
+        logger.info("Checking cluster of kubeconfig context '%s'.", config.context)
+        cluster = await get_k8s_cluster_info(config.kubeconfig, config.context)
+        cluster_branch = cluster.version.branch
+
+        if cluster_branch not in supported_branches:
+            logger.error("The K8s cluster version %s of cluster '%s' is already EOL.", cluster.version, cluster.name)
+
+        # XXX: When should we pass allow_older=True? I don't see it in the standard
+        if check_k8s_version_recency(cluster.version, cve_affected_ranges):
```

Comment:
```suggestion
        if cluster_branch not in supported_branches:
            logger.error("The K8s cluster version %s of cluster '%s' is already EOL.", cluster.version, cluster.name)
        elif check_k8s_version_recency(cluster.version, cve_affected_ranges, allow_older=True):
```

**Issue 1**: `elif` is needed because otherwise confusing output can occur:

```
INFO: Checking cluster of kubeconfig context 'oldoldstable'.
ERROR: The K8s cluster version 1.26.14 of cluster 'oldoldstable' is already EOL.
INFO: The K8s cluster version 1.26.14 of cluster 'oldoldstable' is still in the recency time window.
```

**Issue 2**: The `allow_older` param must be set to `True` always now. Otherwise, for example, a v1.27 cluster would be detected as outdated by the script.

I introduced `allow_older` to able to reuse the already existing `check_k8s_version_recency(…)` function for clusters that do not use the latest K8s branch, which would be the case for `oldstable` and older but not for `stable` according to the abandoned context name scheme. (So it is an conformance test implementation detail and has no corresponding text in the standard.)

Admittedly this is confusing. Root cause is that this function does two things at a time which should be separate: a) checking if a recent patch is applied for an arbitrary K8s branch and b) checking if the newest K8s branch (plus/minus 4 month) cadence is used.